experimental/compactdecimal/src/compactdecimal.rs

```diff
@@ -0,0 +1,406 @@
+            compact_data: DataProvider::<LongCompactDecimalFormatDataV1Marker>::load(
+                data_provider,
+                DataRequest {
+                    locale,
+                    metadata: Default::default(),
+                },
+            )?
+            .take_payload()?
+            .cast(),
+        })
+    }
+
+    icu_provider::gen_any_buffer_constructors!(
+        locale: include,
+        options: skip,
+        error: CompactDecimalError,
+        functions: [
+            Self::try_new_long_unstable,
+            try_new_long_with_any_provider,
+            try_new_long_with_buffer_provider
+        ]
+    );
+
+    /// Formats an integer in compact decimal notation using the default
+    /// precision settings.
+    ///
+    /// The result may have a fractional digit only if it is compact and its
+    /// significand is less than 10. Trailing fractional 0s are omitted.
+    /// ```
+    /// # use icu_compactdecimal::CompactDecimalFormatter;
+    /// # use icu_locid::locale;
+    /// # use writeable::assert_writeable_eq;
+    /// #
+    /// # let short_english = CompactDecimalFormatter::try_new_short_unstable(
+    /// #    &icu_testdata::unstable(),
+    /// #    &locale!("en").into()
+    /// # ).unwrap();
+    /// assert_writeable_eq!(short_english.format(2), "2");
+    /// assert_writeable_eq!(short_english.format(843), "843");
+    /// assert_writeable_eq!(short_english.format(2207), "2.2K");
+    /// assert_writeable_eq!(short_english.format(15_127), "15K");
+    /// assert_writeable_eq!(short_english.format(3_010_349), "3M");
+    /// ```
+    /// The result is the nearest such compact number, with halfway cases-
+    /// rounded towards the number with an even least significant digit.
+    /// ```
+    /// # use icu_compactdecimal::CompactDecimalFormatter;
+    /// # use icu_locid::locale;
+    /// # use writeable::assert_writeable_eq;
+    /// #
+    /// # let short_english = CompactDecimalFormatter::try_new_short_unstable(
+    /// #    &icu_testdata::unstable(),
+    /// #    &locale!("en").into(),
+    /// # ).unwrap();
+    /// assert_writeable_eq!(short_english.format(999_499), "999K");
+    /// assert_writeable_eq!(short_english.format(999_500), "1M");
+    /// assert_writeable_eq!(short_english.format(1650), "1.6K");
+    /// assert_writeable_eq!(short_english.format(1750), "1.8K");
+    /// assert_writeable_eq!(short_english.format(1950), "2K");
+    /// ```
+    pub fn format(&self, value: i64) -> FormattedCompactDecimal<'_> {
+        let unrounded = FixedDecimal::from(value);
+        let log10_type = unrounded.nonzero_magnitude_start();
+        let (mut plural_map, mut exponent) = self.plural_map_and_exponent_for_magnitude(log10_type);
+        let mut significand = unrounded.multiplied_pow10(-i16::from(exponent));
+        if significand.nonzero_magnitude_start() == 0 {
+            significand.half_even(-1);
```

Comment:
Thought: I wonder if we should make `trunc` the default compact decimal rounding mode in ICU4X? People might not like 999,501 of something (likes, views, emails, ...) rounded up to 1M.

Reply:
I agree that this was a questionable default (YouTube uses directed rounding towards 0 everywhere for the very reason you cite).

However, I would rather we aligned with ICU4everythingelse and other existing implementations.

Aside from compatibility/consistency questions, I should note that truncating by hand is much easier (you don’t need the adjustment on round up), so I would rather let people do the easy one themselves and provide the more involved (yet also reasonable) option than the reverse.